### ISSUE TITLE
Bus Error

### ISSUE THREAD

**Issue by gradse** [AlgoMole/MolCRAFT#8]
We I run python train_bfn.py --test_only --no_wandb --ckpt_path ./checkpoints/last.ckpt I have this issue:

/users/PCON0041/yufei/.conda/envs/bfn/lib/python3.9/site-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: dbf9226-mod is an invalid version and will not be supported in a future release
  warnings.warn(
./logs/yufei_bfn_sbdd/official/default/test_outputs already exists, change test_output_dir to ./logs/yufei_bfn_sbdd/official/default/test_outputs_v46
Global seed set to 1234
wandb: Using wandb-core as the SDK backend. Please refer to https://wandb.me/wandb-core for more information.
wandb: WARNING `resume` will be ignored since W&B syncing is set to `offline`. Starting a new run with run id jg2tajt5.
wandb: Tracking run with wandb version 0.18.0
wandb: W&B syncing is set to `offline` in this directory.  
wandb: Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
protein feature dim: 27, ligand feature dim: 13
The config of this process is:
{
    "project_name": "yufei_bfn_sbdd",
    "exp_name": "official",
    "revision": "default",
    "debug": false,
    "no_wandb": true,
    "wandb_resume_id": null,
    "logging_level": "warning",
    "seed": 1234,
    "test_only": true,
    "empty_folder": false,
    "dynamics": {
        "t_min": 0.0001,
        "sigma1_coord": 0.03,
        "beta1": 1.5,
        "protein_atom_feature_dim": 27,
        "ligand_atom_feature_dim": 13,
        "use_discrete_t": true,
        "discrete_steps": 1000,
        "destination_prediction": true,
        "sampling_strategy": "end_back_pmf",
        "node_indicator": true,
        "time_emb_dim": 1,
        "time_emb_mode": "simple",
        "center_pos_mode": "protein",
        "pos_init_mode": "zero",
        "net_config": {
            "name": "unio2net",
            "num_blocks": 1,
            "num_layers": 9,
            "hidden_dim": 128,
            "n_heads": 16,
            "edge_feat_dim": 4,
            "num_r_gaussian": 20,
            "knn": 32,
            "num_node_types": 8,
            "act_fn": "relu",
            "norm": true,
            "cutoff_mode": "knn",
            "ew_net_type": "global",
            "num_x2h": 1,
            "num_h2x": 1,
            "r_max": 10.0,
            "x2h_out_fc": false,
            "sync_twoup": false
        }
    },
    "accounting": {
        "logdir": "./logs/yufei_bfn_sbdd/official/default",
        "dump_config_path": "./logs/yufei_bfn_sbdd/official/default/config.yaml",
        "wandb_logdir": "./logs/yufei_bfn_sbdd/official/default",
        "checkpoint_dir": "./logs/yufei_bfn_sbdd/official/default/checkpoints",
        "generated_mol_dir": "./logs/yufei_bfn_sbdd/official/default/generated_mol",
        "val_outputs_dir": "./logs/yufei_bfn_sbdd/official/default/val_outputs",
        "test_outputs_dir": "./logs/yufei_bfn_sbdd/official/default/test_outputs_v46"
    },
    "data": {
        "name": "pl",
        "path": "./data/crossdocked_v1.1_rmsd1.0_pocket10",
        "split": "./data/crossdocked_pocket10_pose_split.pt",
        "transform": {
            "ligand_atom_mode": "add_aromatic"
        },
        "with_split": true,
        "atom_decoder": "['H', 'C', 'N', 'O', 'F', 'P', 'S', 'Cl'] (len=8)",
        "colors_dic": "['#FFFFFF99', 'C7', 'C0', 'C3', 'C1', 'C4', 'C8', 'C9'] (len=8)",
        "radius_dic": "[0.3, 0.6, 0.6, 0.6, 0.6, 0.6, 0.6, 0.6] (len=8)",
        "normalizer_dict": {
            "pos": 2.0
        }
    },
    "visual": {
        "save_mols": true,
        "visual_nums": 10,
        "visual_chain": false
    },
    "train": {
        "batch_size": 8,
        "num_workers": 40,
        "pos_noise_std": 0,
        "random_rot": false,
        "val_freq": 1000,
        "epochs": 15,
        "resume": false,
        "v_loss_weight": 1,
        "ckpt_freq": 1,
        "max_grad_norm": "Q",
        "optimizer": {
            "type": "adam",
            "lr": 0.0005,
            "weight_decay": 0,
            "beta1": 0.95,
            "beta2": 0.999
        },
        "scheduler": {
            "type": "plateau",
            "factor": 0.6,
            "patience": 10,
            "min_lr": 1e-06,
            "max_iters": 187485
        },
        "ema_decay": 0.999
    },
    "evaluation": {
        "ckpt_path": "./checkpoints/last.ckpt",
        "batch_size": 100,
        "sample_steps": 100,
        "num_samples": 5,
        "sample_num_atoms": "ref",
        "docking_config": {
            "mode": "vina_score",
            "protein_root": "./data/test_set",
            "exhaustiveness": 16
        },
        "ligand_path": "null",
        "protein_path": "null"
    }
}
/users/PCON0041/yufei/.conda/envs/bfn/lib/python3.9/site-packages/lightning_lite/plugins/environments/slurm.py:167: PossibleUserWarning: The `srun` command is available on your system but is not used. HINT: If your intention is to run Lightning on SLURM, prepend your python command with `srun` like so: srun python train_bfn.py --test_only --no_wandb --ckpt_path ./ch ...
  rank_zero_warn(
/users/PCON0041/yufei/.conda/envs/bfn/lib/python3.9/site-packages/lightning_lite/accelerators/cuda.py:159: UserWarning: Can't initialize NVML
  warnings.warn("Can't initialize NVML")
GPU available: False, used: False
/users/PCON0041/yufei/.conda/envs/bfn/lib/python3.9/site-packages/pkg_resources/__init__.py:116: PkgResourcesDeprecationWarning: dbf9226-mod is an invalid version and will not be supported in a future release
  warnings.warn(
/users/PCON0041/yufei/.conda/envs/bfn/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/version.py:111: DeprecationWarning: Creating a LegacyVersion has been deprecated and will be removed in the next major release
  warnings.warn(
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
Restoring states from the checkpoint path at ./checkpoints/last.ckpt
Loaded model weights from checkpoint at ./checkpoints/last.ckpt
/users/PCON0041/yufei/.conda/envs/bfn/lib/python3.9/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:224: PossibleUserWarning: The dataloader, test_dataloader 0, does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` (try 40 which is the number of cpus on this machine) in the `DataLoader` init to improve performance.
  rank_zero_warn(
Testing: 0it [00:00, ?it/s]Bus error

I think this may relate to PyTorch Lightning but don't know how to solve it, does anyone encounter this error?

**Comment by qky18** [comment]
Hi, sorry for the confusion caused. I've checked the `environment.yml` file and found that the `pytorch_lightning` version is wrongly set to 1.8.4, while it should be 2.0.8.

**Comment by qky18** [comment]
Hi, if there's no further discussion, I will close the issue in 24 hrs.
Feel free to reopen it if the problem persists.
